The job **`pr-check-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/16223475904/job/45809694790) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/45809694790) [(plain)](https://github.com/rust-lang/rust/commit/3ed08f8ee669d459f80c4e706fd2b545ec8073f8/checks/45809694790/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling tracing-forest v0.1.6
   Compiling tracing-chrome v0.7.2
   Compiling opener v0.5.2
   Compiling build_helper v0.1.0 (/checkout/src/build_helper)
error[E0432]: unresolved import `$crate::utils::exec::FormatShortCmd`
##[error]    --> src/bootstrap/src/utils/tracing.rs:55:17
     |
55   |             use $crate::utils::exec::FormatShortCmd;
     |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `FormatShortCmd` in `utils::exec`
     |
    ::: src/bootstrap/src/core/build_steps/compile.rs:2530:21
     |
2530 |     let _run_span = crate::trace_cmd!(cmd);
     |                     ---------------------- in this macro invocation
     |
     = note: this error originates in the macro `crate::trace_cmd` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0432]: unresolved import `$crate::utils::exec::FormatShortCmd`
##[error]   --> src/bootstrap/src/utils/tracing.rs:55:17
    |
55  |             use $crate::utils::exec::FormatShortCmd;
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `FormatShortCmd` in `utils::exec`
    |
   ::: src/bootstrap/src/utils/exec.rs:700:25
    |
700 |         let _run_span = trace_cmd!(command);
    |                         ------------------- in this macro invocation
    |
    = note: this error originates in the macro `trace_cmd` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0624]: method `format_short_cmd` is private
##[error]    --> src/bootstrap/src/utils/tracing.rs:61:42
     |
61   |                 cmd = $cmd.fingerprint().format_short_cmd(),
     |                                          ^^^^^^^^^^^^^^^^ private method
     |
    ::: src/bootstrap/src/utils/exec.rs:82:5
     |
82   |     fn format_short_cmd(&self) -> String {
     |     ------------------------------------ private method defined here
     |
    ::: src/bootstrap/src/core/build_steps/compile.rs:2530:21
     |
2530 |     let _run_span = crate::trace_cmd!(cmd);
     |                     ---------------------- in this macro invocation
     |
     = note: this error originates in the macro `crate::trace_cmd` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0432, E0624.
For more information about an error, try `rustc --explain E0432`.
error: could not compile `bootstrap` (lib) due to 3 previous errors
failed to run: /checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo build --manifest-path /checkout/src/bootstrap/Cargo.toml -Zroot-dir=/checkout --features=tracing --locked --features build-metrics
  local time: Fri Jul 11 16:05:30 UTC 2025
  network time: Fri, 11 Jul 2025 16:05:30 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/bootstrap/src/utils/exec.rs
+++ b/src/bootstrap/src/utils/exec.rs
@@ -76,8 +76,10 @@ pub struct CommandFingerprint {
     cwd: Option<PathBuf>,
 }
 
-impl FormatShortCmd for CommandFingerprint {
-    fn format_short_cmd(&self) -> String {
+impl CommandFingerprint {
+    /// Helper method to format both Command and BootstrapCommand as a short execution line,
+    /// without all the other details (e.g. environment variables).
+    pub fn format_short_cmd(&self) -> String {
         let program = Path::new(&self.program);
         let mut line = vec![program.file_name().unwrap().to_str().unwrap().to_owned()];
         line.extend(self.args.iter().map(|arg| arg.to_string_lossy().into_owned()));
@@ -542,29 +544,6 @@ impl Default for CommandOutput {
             stdout: Some(vec![]),
             stderr: Some(vec![]),
         }
-    }
-}
-
-/// Helper trait to format both Command and BootstrapCommand as a short execution line,
-/// without all the other details (e.g. environment variables).
-pub trait FormatShortCmd {
-    fn format_short_cmd(&self) -> String;
-}
-
-#[cfg(feature = "tracing")]
-impl FormatShortCmd for BootstrapCommand {
-    fn format_short_cmd(&self) -> String {
-        self.command.format_short_cmd()
-    }
-}
-
-#[cfg(feature = "tracing")]
-impl FormatShortCmd for Command {
-    fn format_short_cmd(&self) -> String {
-        let program = Path::new(self.get_program());
-        let mut line = vec![program.file_name().unwrap().to_str().unwrap()];
-        line.extend(self.get_args().map(|arg| arg.to_str().unwrap()));
-        line.join(" ")
     }
 }
 

--- a/src/bootstrap/src/utils/tracing.rs
+++ b/src/bootstrap/src/utils/tracing.rs
@@ -52,8 +52,6 @@ macro_rules! error {
 macro_rules! trace_cmd {
     ($cmd:expr) => {
         {
-            use $crate::utils::exec::FormatShortCmd;
-
             ::tracing::span!(
                 target: "COMMAND",
                 ::tracing::Level::TRACE,

